tests/src/JIT/Performance/CodeQuality/Span/SpanBench.cs

```diff
@@ -781,14 +1028,33 @@ public static void TestSpanAsBytesByte(int length)
         [InlineData(1000)]
         public static void TestSpanAsBytesInt(int length)
         {
-            var array = new int[length];
-            var span = new Span<int>(array);
-            foreach (var iteration in Benchmark.Iterations)
-                using (iteration.StartMeasurement())
-                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
-                    {
-                        Span<byte> temp = span.AsBytes<int>();
-                    }
+            InvokeTestSpanAsBytes<int>(length);
+        }
+
+        static void InvokeTestSpanAsBytes<T>(int length)
+            where T : struct
+        {
+            var array = new T[length];
+
+            Invoke((int innerIterationCount) => TestSpanAsBytes<T>(array, innerIterationCount, false),
+                "TestSpanAsBytes<{0}>({1})", typeof(T).Name, length);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static void TestSpanAsBytes<T>(T[] array, int iterationCount, bool untrue)
+            where T : struct
+        {
+            var sink = Sink<byte>.Instance;
+            var span = new Span<T>(array);
+
+            for (int i = 0; i < iterationCount; i++)
+            {
+                var byteSpan = span.AsBytes();
+                // Under a condition that we know is false but the jit doesn't,
+                // add a read from 'byteSpan' to make sure it's not dead, and an assignment
+                // to 'span' so the AsBytes call won't get hoisted.
+                if (untrue) { sink.Data = byteSpan[0]; span = new Span<T>(); }
```

Comment:
This is interesting. I get the read from 'byteSpan', but why do we need to assign to 'span'? If we don't, will the call to AsBytes gets hoisted outside of the for loop and will only run once? Is that what "hoisted" means?